llpc/translator/lib/SPIRV/SPIRVReader.cpp

```diff
@@ -6336,6 +6336,32 @@ Value *SPIRVToLLVM::transSPIRVImageQueryLodFromInst(SPIRVInstruction *bi, BasicB
     result = ConvertingSamplerSelectLadderHelper(result, convertingSamplerIdx, createImageGetLod);
   }
 
+  // NOTE: This is a workaround. When UV width equals 0, the result return 0, but we expect the value is
+  // -22.0 to -infinity.
+  // Derivative operations are performed on the XY of UV coordinate respectively, if both of result
+  // are zero, we will return -FLT_MAX for LOD.
+  Value *dpdx = getBuilder()->CreateDerivative(coord, false, false);
+  Value *dpdy = getBuilder()->CreateDerivative(coord, true, false);
+  Value *absDpdx = getBuilder()->CreateUnaryIntrinsic(Intrinsic::fabs, dpdx);
+  Value *absDpdy = getBuilder()->CreateUnaryIntrinsic(Intrinsic::fabs, dpdy);
+  Value *add = getBuilder()->CreateFAdd(absDpdx, absDpdy);
+  Value *cmp = getBuilder()->CreateFCmpOEQ(add, ConstantFP::get(coord->getType(), 0.0));
+
+  Value *con = getBuilder()->getInt1(true);
+  if (cmp->getType()->isVectorTy()) {
+    auto vec = cast<FixedVectorType>(cmp->getType());
```

Comment:
Could merge these two statements
for (unsigned elemIdx = 0; elemIdx < cast<FixedVectorType>(maybeZero->getType())->getNumElements(); elemIdx++) {

Reply:
Fixed.